apps/entity-service/src/entities/processors/association-processor.ts

```diff
@@ -53,20 +65,39 @@ export abstract class AssociationProcessor<M extends UuidModel<M>, D extends Ass
     return this._baseEntity;
   }
 
-  async addDtos(document: DocumentBuilder): Promise<void> {
-    const associations = await this.getAssociations(await this.getBaseEntity());
+  async addDtos(document: DocumentBuilder, query?: MediaQueryDto): Promise<void> {
+    const associations = await this.getAssociations(await this.getBaseEntity(), query);
 
     const additionalProps = { entityType: this.entityType, entityUuid: this.entityUuid };
     const indexIds: string[] = [];
     for (const association of associations) {
       indexIds.push(association.uuid);
-      document.addData(association.uuid, new this.DTO(association, additionalProps));
+      if (this.DTO.name === MediaDto.name) {
```

Comment:
I definitely don't want to see custom code specific to one type of association DTO in the processor super class. Anything custom like this should be handled in the custom subclass.

Reply:
removed as it wasn't needed after all